app/src/main/java/com/geeksville/mesh/MainActivity.kt

```diff
@@ -406,6 +400,26 @@ class MainActivity : AppCompatActivity(), Logging,
         askToRate()
     }
 
+    private fun updateConnectionStatusImage(connected: MeshService.ConnectionState) {
+
+        val (image, tooltip) = when (connected) {
+            MeshService.ConnectionState.CONNECTED -> Pair(R.drawable.cloud_on, R.string.connected)
+            MeshService.ConnectionState.DEVICE_SLEEP -> Pair(
+                R.drawable.ic_twotone_cloud_upload_24,
+                R.string.device_sleeping
+            )
+            MeshService.ConnectionState.DISCONNECTED -> Pair(
+                R.drawable.cloud_off,
+                R.string.disconnected
+            )
+            else -> Pair(R.drawable.cloud_off, R.string.disconnected)
+        }
+
+        connectStatusImage.setImageResource(image)
+        TooltipCompat.setTooltipText(connectStatusImage, getString(tooltip))
+    }
+
```

Comment:
Your code is super clean and nice.  I especially like how you hoisted the two bits of state out of the when.  A very 'kotlin' approach!

Reply:
Thanks

I tried something easy to get familiar with the project and github process. In fact, I did not like kotlin when I followed a tutorial for the Jetpack menu navigation. This project is perfect to change my mind I hope !